### ISSUE TITLE
Unable to generate images in tensor board using keras (showing only black and grey coloured images)

### ISSUE THREAD

**Issue by NaveenGandla** [tensorflow/tensorboard#2058]
- TensorBoard version (from pip package, also printed out when running `tensorboard`)= 1.12
 - TensorFlow version if different from TensorBoard = tensorflow-gpu 1.12 
 - OS Platform and version = Linux Ubuntu 16.04)
 - Python version (e.g. 2.7, 3.5) = 3.6 
 - For browser-related issues:
   - Browser type and version = Chrome 64.0.3282.140)
   - Screenshot if it's a visual issue
 
![tensor_board_problem](https://user-images.githubusercontent.com/44918507/55000222-1687e800-4ff8-11e9-91f9-8a8d1423cb47.png)

my code : 
```py
[from tensorflow.keras.callbacks import TensorBoard
tensorboard = TensorBoard(log_dir="/home/lebenpc/Messidor/cnn_logs", histogram_freq=1, write_graph=False, write_images=True)
history = model.fit(trainX,trainY,epochs=10,batch_size=8,validation_data=(testX,testY),callbacks=[tensorboard])](url)
```

**Comment by stephanwlee** [comment]
Hi @NaveenGandla, can you provide a bit more detail on what your model looks like? A full example will allow us to debug it better. Thanks!

**Comment by shashvatshahi1998** [comment]
import tensorflow as tf

def make_image(tensor):
    """
    Convert an numpy representation image to Image protobuf.
    Copied from https://github.com/lanpa/tensorboard-pytorch/
    """
    from PIL import Image
    height, width, channel = tensor.shape
    image = Image.fromarray(tensor)
    import io
    output = io.BytesIO()
    image.save(output, format='PNG')
    image_string = output.getvalue()
    output.close()
    return tf.Summary.Image(height=height,
                         width=width,
                         colorspace=channel,
                         encoded_image_string=image_string)

class TensorBoardImage(keras.callbacks.Callback):
    def __init__(self, tag):
        super().__init__() 
        self.tag = tag

    def on_epoch_end(self, epoch, logs={}):
        # Load image
        img = data.astronaut()
        # Do something to the image
        img = (255 * skimage.util.random_noise(img)).astype('uint8')

        image = make_image(img)
        summary = tf.Summary(value=[tf.Summary.Value(tag=self.tag, image=image)])
        writer = tf.summary.FileWriter('./logs')
        writer.add_summary(summary, epoch)
        writer.close()

        return

tbi_callback = TensorBoardImage('Image Example')

This is the common skeleton used for displaying image on Tensorboard.
Just pass the callback to fit or fit_generator.

**Comment by shashvatshahi1998** [comment]
or you may also try reinstalling tensorboard (using tf nightly)

**Comment by stephanwlee** [comment]
AFAICT, it looks like the color channel is properly used only for 4D tensors: https://github.com/tensorflow/tensorflow/blob/22653fa1ac370a8d537dc8ee610d26926d955436/tensorflow/python/keras/callbacks.py#L1435-L1436

@omalleyt12 Is this working as intended?

**Comment by NaveenGandla** [comment]
@stephanwlee   
this is how my model looks like 
model = Sequential()
model.add(Conv2D(16, 5, activation='relu', padding='same', input_shape=ip))
model.add(Conv2D(16, 5, activation='relu',padding='same'))
model.add(BatchNormalization(momentum=0.99,epsilon=0.001))
model.add(MaxPooling2D(2))
model.add(Conv2D(32, 5, activation='relu', padding='same'))
model.add(Conv2D(32, 5, activation='relu',padding='same'))
model.add(BatchNormalization(momentum=0.99,epsilon=0.001))
model.add(MaxPooling2D(2))
model.add(Conv2D(64, 5, activation='relu', padding='same'))
model.add(Conv2D(64, 5, activation='relu',padding='same'))
model.add(BatchNormalization(momentum=0.99,epsilon=0.001))
model.add(MaxPooling2D(2))
model.add(Conv2D(128, 5, activation='relu', padding='same'))
model.add(Conv2D(128, 5, activation='relu',padding='same'))
model.add(BatchNormalization(momentum=0.99,epsilon=0.001))
model.add(MaxPooling2D(2))
model.add(Conv2D(256, 5, activation='relu', padding='same'))
model.add(Conv2D(256, 5, activation='relu',padding='same'))
model.add(BatchNormalization(momentum=0.99,epsilon=0.001))
model.add(MaxPooling2D(2))
model.add(Conv2D(512, 5, activation='relu', padding='same'))
model.add(Conv2D(512, 5, activation='relu',padding='same'))
model.add(BatchNormalization(momentum=0.99,epsilon=0.001))
model.add(MaxPooling2D(2)
model.add(Flatten())
model.add(Dropout(0.25))
model.add(Dense(1024, activation='relu'))
model.add(Dropout(0.5))
model.add(Dense(1, activation='sigmoid'))


Sample input images : 
![20051020_44762_0100_PP tif_rot30](https://user-images.githubusercontent.com/44918507/55310869-2895da80-547f-11e9-9d16-8a6074a6da34.JPG)
![20051020_44782_0100_PP tif_rot30](https://user-images.githubusercontent.com/44918507/55310871-292e7100-547f-11e9-9a60-33e753240ebc.JPG)
![20051020_44843_0100_PP tif_rot30](https://user-images.githubusercontent.com/44918507/55310872-292e7100-547f-11e9-8b08-11b9726ae917.JPG)

**Comment by wchargin** [comment]
Hi @NaveenGandla—please provide a complete, self-contained Python file
that we can run to reproduce the problem.

**Comment by hgadig** [comment]
Closing as this issue is in "awaiting response" status for more than 3 days. Please provide the information asked above and we will reopen. Thank you!
